plugins/quantum_k_nearest_neighbours/frontend_js/__init__.py

```diff
@@ -0,0 +1,26 @@
+# Copyright 2023 QHAna plugin runner contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+_file_dir = os.path.dirname(__file__)
+
+_frontend_js_content = ""
+with open(_file_dir + "/frontend.js") as f:
```

Comment:
Before, the file contents were read once into memory and were served from memory every time (this could be considered a performance optimization). Now, the file is read for each request. In this case it is better to use the correct method of the flask framework to serve the file: <https://flask.palletsprojects.com/en/2.2.x/api/#flask.send_file>
If you want to keep the optimization, then move reading in the file out of the method again (or access the module scope from the method to store the read in value in a module variable, e.g., with the `global` keyword)

Reply:
I opted to use flask's send_file method.